automation/functions/script.ps1

```diff
@@ -1,26 +1,38 @@
-param(
+ param(
     [Parameter(Mandatory = $True)]
     [string]
-    $resourceGroupName,
+    $SubscriptionId,
 
     [Parameter(Mandatory = $True)]
     [string]
-    $location
+    $TenantId,
+
+    [Parameter(Mandatory = $True)]
+    [string]
+    $ServicePrincipalId,
+
+    [Parameter(Mandatory = $True)]
+    [string]
+    $ServicePrincipalPassword
 )
 
-#sign into azure account 
-Connect-AzAccount
+
+#sign into azure account
+$securePassword = ConvertTo-SecureString -String $ServicePrincipalPassword -AsPlainText -Force;
+$credentials = New-Object -TypeName System.Management.Automation.PSCredential($ServicePrincipalId, $securePassword);
+Connect-AzAccount -Credential $credentials -Tenant $TenantId -ServicePrincipal
```

Comment:
This will not always work. If the account has multiple subscriptions you will not be sure which one will get selected.